Added a note that clarifies trigger recommendation for Docker repositories (#4881)
harness/developer-hub: docs/platform/triggers/trigger-on-a-new-artifact.md

- Do not trigger on the **latest** tag of an artifact, such as a Docker image. With latest, Harness only has metadata, such as the tag name, which has not changed, and so Harness does not know if anything has changed. The trigger will not be executed.
- In Harness, you can select who is able to create and use triggers within Harness, but you must use your repository's RBAC to control who can add the artifacts or initiate the events that start the Harness trigger.
- Whenever you create a trigger for the first time, Harness recommends submitting a tag or pushing an artifact to verify its functionality. By doing this, the trigger will execute and the pipeline will run as expected when subsequent tags are pushed.

  :::note

>   In a Docker repository linked to a trigger, the trigger initially stays 'pending' since there are no available tags. Upon the first artifact push, the trigger status changes to 'success' because of new tags, yet this alone won't activate the pipeline. **The pipeline is triggered only after a second push to Docker.**

Reviewer: ```suggestion
  When you link a Docker repository to a trigger, the trigger status will remain `pending` until there are available tags. After the first artifact push, the trigger status changes to `success` because of new tags, but this alone will not activate the pipeline. **The pipeline will only be triggered after a second push to Docker.**
```
